glass/lensing.py

```diff
@@ -224,6 +224,79 @@ def w(zi, zj, zk):
         delta12 = delta23
 
 
+@generator('zmin, zmax, delta, kappa -> kappa')
+def ia_nla(cosmo, A_ia, eta=0., z0=0., lbar=0., l0=1e-9, beta=0.):
+    r'''Implementation of intrinsic alignments using the NLA model
+
+    Notes
+    -----
+    The instrinsic alignments convergence :math:`\kappa_{\rm{IA}}` is computed
+    from the matter overdensity field :math:`\delta` using the Non-linear
+    Alignments Model (NLA) [1, 2, 3]_ in the following manner.
+
+    We define the intrinsic alignment amplitude :math:`\f_{\rm{NLA}}` for a given
+    redshift shell at its midpoint in comoving distance at :math:`z` as
+
+    .. math::
+
+        \f_{\rm{NLA}} = - A_{\rm{IA}} \frac{C_{1} \overline{\rho}(z)}{\overline{D}(z)}
+        \bigg(\frac{1+z}{1+z_{0}}\bigg)^{\eta}
+        \bigg(\frac{\overline{L}}{L_{0}}\bigg)^{\beta}  \;.
+
+    where :math:`A_{\rm{IA}}` is the intrinsic alignments amplitude, :math:`C_{1}` is a
+    normalisation constant set to :math:`C_{1} = 5 \times 10^{-14} (h^{2} M_{\odot}/Mpc^{-3})^{-2}` [2]_,
+    :math:`\overline{\rho}` is the mean matter density of the Universe at redshift :math:`z`,
+    :math:`\overline{D}(z)` is the growth factor, :math:`\eta` is the power of the power law
+    which describes the redshift-dependence of the IA with respect to :math:`z_{0}` and
+    :math:`\beta` is the power of the power law which describes the dependence on luminosity,
+    :math:`\overline{L}`, of the IA with respect to :math:`L_{0}`.
+
+    Then, the convergence of the matter shells is updated by adding the contribution from IAs
+    to the convergence from lensing as follows
+
+    .. math::
+
+        \kappa(z) = \kappa_{\rm{lensing}}(z) + \f_{\rm{NLA}}(z, \overline{L}) \delta(z)
+
+    where :math:`\delta(z)` is the matter overdensity field at redshift :math:`z`.
+
+    References
+    ----------
+    .. [1] Catelan et al., 2000
+    .. [2] Hirata & Seljak, 2004
+    .. [3] Bridle & King, 2007
```

Comment:
People use these references to write their papers, so they should be complete e.g. in MNRAS format.

You can go to the ADS page for these papers, then export the citation in MNRAS format, and paste that here (modulo some cleanup).